src/WebJobs.Script/Workers/SharedMemoryDataTransfer/SharedMemoryManager.cs

```diff
@@ -0,0 +1,314 @@
+            sharedMemoryMaps.Add(mapName);
+        }
+
+        public async Task<object> GetObjectAsync(string mapName, int offset, int count, Type objectType)
+        {
+            if (objectType == typeof(byte[]))
+            {
+                return await GetBytesAsync(mapName, offset, count);
+            }
+            else if (objectType == typeof(string))
+            {
+                return await GetStringAsync(mapName, offset, count);
+            }
+
+            return null;
+        }
+
+        public bool TryFreeSharedMemoryMapsForInvocation(string invocationId)
+        {
+            if (!InvocationSharedMemoryMaps.TryRemove(invocationId, out HashSet<string> mapNames))
+            {
+                _logger.LogTrace("No shared memory maps allocated for invocation id: {Id} by the host", invocationId);
+                return true;
+            }
+
+            bool freedAll = true;
+            int numFreed = 0;
+            foreach (string mapName in mapNames)
+            {
+                if (TryFreeSharedMemoryMap(mapName))
+                {
+                    numFreed++;
+                }
+                else
+                {
+                    freedAll = false;
+                }
+            }
+
+            _logger.LogTrace("Freed shared memory maps for invocation id: {Id} - Count: {Count}, FreedAll: {FreedAll}", invocationId, numFreed, freedAll);
+            return freedAll;
+        }
+
+        public bool TryFreeSharedMemoryMap(string mapName)
+        {
+            if (!AllocatedSharedMemoryMaps.TryRemove(mapName, out SharedMemoryMap sharedMemoryMap))
+            {
+                _logger.LogTrace("Cannot find SharedMemoryMap: {mapName}", mapName);
+                return false;
+            }
+
+            sharedMemoryMap.Dispose();
+            return true;
+        }
+
+        public bool IsSupported(object input)
+        {
+            if (input is byte[] arr)
+            {
+                int arrBytes = arr.Length;
+                if (arrBytes >= SharedMemoryConstants.MinObjectBytesForSharedMemoryTransfer && arrBytes <= SharedMemoryConstants.MaxObjectBytesForSharedMemoryTransfer)
+                {
+                    return true;
+                }
+
+                _logger.LogDebug("Cannot transfer bytes over shared memory; size {Size} less than minimum threshold", arrBytes);
+                return false;
+            }
+            else if (input is string str)
+            {
+                int strBytes = str.Length * sizeof(char);
+                if (strBytes >= SharedMemoryConstants.MinObjectBytesForSharedMemoryTransfer && strBytes <= SharedMemoryConstants.MaxObjectBytesForSharedMemoryTransfer)
+                {
+                    return true;
+                }
+
+                _logger.LogTrace("Cannot transfer string over shared memory; size {Size} less than minimum threshold", strBytes);
```

Comment:
nit: Fix the log statement to indicate  size is not supported. It is either less than min threshold or greater max threshold.